### PR TITLE
std.Target.Query: fix compilation error

### LINKED ISSUE #21668: canDetectLibC should check target Target.abi
### Zig Version

0.14.0-dev.1827+e1e151df0

### Steps to Reproduce and Observed Behavior

```
$zig libc
```
Error:
bug1

```
/Users/dk/src/zig/zig/lib/std/Target/Query.zig:377:38: error: no field or member function named 'isAndroid' in '?Target.Abi'
        if (os == .linux and self.abi.isAndroid()) return true;
                             ~~~~~~~~^~~~~~~~~~
referenced by:
    main: /Users/dk/src/zig/zig/lib/compiler/libc.zig:116:40
    main: /Users/dk/src/zig/zig/lib/std/start.zig:621:37
    3 reference(s) hidden; use '-freference-trace=5' to see all references
```

### Expected Behavior

should not be error

### PR DESCRIPTION
**Pull request by dravenk**
closes #21668

### CODE REVIEW

File: lib/std/Target/Query.zig [hunk 1]
```diff
@@ -374,7 +374,7 @@ pub fn canDetectLibC(self: Query) bool {
     if (self.isNativeOs()) return true;
     if (self.os_tag) |os| {
         if (builtin.os.tag == .macos and os.isDarwin()) return true;
-        if (os == .linux and self.abi.isAndroid()) return true;
+        if (os == .linux and self.abi.?.isAndroid()) return true;
```
[the remark this answers is not in the document]
Comment (andrewrk) [reply lib/std/Target/Query.zig:377]: Please avoid `.?` syntax altogether in this case.

[the remark this answers is not in the document]
Comment (dravenk) [reply lib/std/Target/Query.zig]: > Please avoid `.?` syntax altogether in this case.
Refer to system.zig, completely ignore.? syntax.
Reference: https://github.com/ziglang/zig/blob/13fb68c064900cee67583e603d8c2acf41770663/lib/std/zig/system.zig#L976

**Review by alexrp — APPROVED** [review]

**Review by andrewrk — CHANGES_REQUESTED** [review]

**Comment by alexrp** [comment]
This isn't strictly related to this PR as it's just restoring existing behavior, but I don't understand why Android actually gets special-cased here. The idea seems to be that you can do `CC=/path/to/android/sdk/cc zig libc -target arm-linux-androideabi` or similar. But why is Android special in this regard? Surely I could just as well do `CC=mips64-linux-gnuabi64-gcc zig libc -target mips64-linux-gnuabi64`? cc @mikdusan

**Comment by mikdusan** [comment]
@alexrp I added this a while back leaning towards a curated approach of non-native libc detection and so checks were added for the only 2 platforms I tested against: various apple SDKs and android NDK. Also this impl doesn't explicitly allow NDK on macos or windows hosts, and although rare, the NDKs are available.

What I was happy about is that a garbage libc.txt was avoided. But if we take out `canDetectLibC()` (non-native logic), everything just falls through to `LibCInstallation.findNative()` and useless libc.txt is generated, or a very unhelpful error is given such as `LibCRuntimeNotFound` in case of `zig libc -target aarch64-linux-android` when NOT setting CC to the correct NDK pathname. We were getting issues raised against this.

In a better impl, what I'd like to see is some tech where libc.txt paths are fuzzy-validated.

To be clear, I think it is important to support exactly the use case pattern you listed (general cross compiler pathname) and it could very well be the curated approach is at odds with that goal.

**Pull request merged**